packages/frontend-2/middleware/requireValidProject.ts

```diff
@@ -19,17 +19,46 @@ export default defineNuxtRouteMiddleware(async (to) => {
       variables: { id: projectId },
       context: {
         skipLoggingErrors: true
-      }
+      },
+      fetchPolicy: 'network-only'
     })
     .catch(convertThrowIntoFetchResult)
 
-  // If project succesfully resolved, move on
-  if (data?.project?.id) return
+  // If project is public or link shareable, allow access regardless of SSO
+  if (
+    data?.project?.visibility === ProjectVisibility.Public ||
+    data?.project?.visibility === ProjectVisibility.Unlisted
+  ) {
+    return
+  }
+
+  // Check for SSO session error
+  const ssoSessionError = (errors || []).find(
+    (e) => e.extensions?.['code'] === 'SSO_SESSION_MISSING_OR_EXPIRED_ERROR'
```

Comment:
Is there a constant or enum you can use to get these codes? Now if you are not familiar with the code it seems like a random string

Reply:
No enum existed from the backend, but I've added one in types.ts that covers all the SSO error codes we receive from the API.